src/lib/storages/RedisProxyCache.ts

```diff
@@ -181,11 +246,60 @@ export class RedisProxyCache implements IProxyCache {
     }
   }
 
+  private async processNode(node: Redis, options: ProcessNodeOptions): Promise<void> {
+    const { pattern: match, batchSize: count, callbackFn, resolve, reject } = options
+    const stream = node.scanStream({ match, count })
+
+    stream.on('data', async (keys) => {
+      stream.pause()
+      try {
+        await Promise.all(keys.map((key: string) => this.processKey(key, callbackFn)))
+      } catch (err: unknown) {
+        stream.destroy(err as Error)
+        reject(err)
+      }
+      stream.resume()
+    })
+    stream.on('end', resolve)
+  }
+
+  private async processKey(key: string, callbackFn: ProcessKeyCallback): Promise<any> {
+    const actualKey = key.replace(':expiresAt', '')
+    const expiresAt = await this.redisClient.get(key)
+
+    if (Number(expiresAt) >= Date.now()) return
+    
+    try {
+      await callbackFn(actualKey)
```

Comment:
2 questions:

1. Maybe, we should add `.catch()` block to _callbackFn_? Otherwise in case of error in _callbackFn_ for this key, we won't be able to delete the key.
2. What if _callbackFn_ takes some time, and during this time we get `receivedSuccessResponse / receivedErrorResponse` call from proxy.  So maybe, to avoid sending 2 callbacks, we need to trigger _callbackFn_ and _key deletion_ simultaneously? E.g.:
```
const deletion =  this.isCluster
   ? () => Promise.all([  this.redisClient.del(actualKey),  this.redisClient.del(key) ])
   : () => this.executePipeline([ ['del', actualKey], ['del', key] ])
 
 await Promise.all([
    callbackFn(actualKey).catch(... // log error),
    deletion(),
 ]);
```

Reply:
@geka-evk 

1. There is no need for additional `.catch()` for `callbackFn` as it is already wrapped in a `try..catch`. The idea is not to delete the keys if there is an error with the callback so it can be retried in the next cycle. 
As you can see in my comment, a design decision needs to be made on when to give up on a key (e.g a second but much higher expiry timeframe). 

2a. 

> What if callbackFn takes some time, and during this time we get receivedSuccessResponse / receivedErrorResponse call from proxy.

What if we don't get `receivedSuccessResponse / receivedErrorResponse` and the callbackFn fails the first time (DFSP is temporary down) but succeeded the second time and we have already deleted the keys?

2b. 

> So maybe, to avoid sending 2 callbacks, we need to trigger callbackFn and key deletion simultaneously?

Unless we have a proper concurrency control per key, deleting both keys regardless of the result of the callback will do more harm than good.